src/App/Search/ReindexSearchClient.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Search;
+
+use App\Model\ModelInterface;
+use App\Search\Mapping\FrameworkMapping;
+use App\Search\Mapping\SupplierMapping;
+use App\Search\FrameworkSearchClient;
+use App\Search\SupplierSearchClient;
+use Elastica\Mapping;
+use Elastica\Reindex;
+use Elastica\Client;
+
+class ReindexSearchClient extends AbstractSearchClient
+{
+    public function reindex(string $indexName, array $analysis)
```

Comment:
This doesn't look right. Why isn't this function in the AbstractSearchClient itself and only extended in the child class if there are changes to the base behaviour?

Additionally we should likely look at the getIndexOrCreate() function to see if it should be modified to include this behaviour if we are intending to do this at every run.
 
If it is not recommended to reindex each time, we could potentially do it based on a conditional environment variable that is set externally.

Reply:
**_This doesn't look right. Why isn't this function in the AbstractSearchClient itself and only extended in the child class if there are changes to the base behaviour?_**
You're right, initially I my code didn't allow for this as I had some conditions in the ReindexSearchClient constructer that was different from AbstractSearchClient. But my refactor now only has 2 functions in there and nothing else so I will change this and add to AbstractSearchClient.

**_Additionally we should likely look at the getIndexOrCreate() function to see if it should be modified to include this behaviour if we are intending to do this at every run._**
I think we should leave this alone as it's called multiple times with a loop when we add or update document to the index. Adding a reindex function or modifying the behaviour of this could have unintended consequences especially if the index is accidentally deleted or lost somehow. 

**_If it is not recommended to reindex each time, we could potentially do it based on a conditional environment variable that is set externally._**

Yes, I think this would a good approach. Where would have the variable, however? The .env file? bear in mind that we can only change the webdev .env file ourselves. Do you have any suggestions for this?